### ISSUE TITLE
Feature Request: DS3 Wired/Bluetooth Support

### ISSUE THREAD

**Issue by DrexionVzanphon** [Jays2Kings/DS4Windows#105]
After using this application I find it much better than Scptoolkit or DS3 tool in pretty much every way. I would love to see the ability to integrate Dual Shock 3 support as the biggest setback it that with Scptoolkit you have to dedicate the BT dongle to only work with that one device. I am sure there is a better way to do it so that you can choose which controller to use with DS4Windows.

**Comment by Bluebie** [comment]
the dualshock 3 uses bluetooth and usb hid in nonstandard ways, and supporting them requires low level hacks. On most unix systems libusb is able to talk over USB with the controller to activate it, but due to bluetooth's more propritary nature bluetooth stacks tend to be propritary and locked up. On OS X DS3 support ended up being something apple added in to their kernel as a specific compatibility hack. Linux support is similar. Without the assistance of microsoft, it's unlikely DS3 support on windows will ever be simple and userspace. DS3 drivers work by talking directly to a USB bluetooth radio and preventing the operating system from attaching it's own bluetooth stack. By reimplementing the gamepad-relevent parts of bluetooth themselves, they can do the hacky stuff needed, but there's no way to share that radio with the OS for more generic bluetooth support.

TLDR: probably impossible with current system architecture, unless someone at Microsoft hears our pleas and implements hacks in the standard microsoft bluetooth stack in the same way Apple did for OS X.
